vllm/attention/backends/flash_attn.py

```diff
@@ -18,6 +17,129 @@
 if TYPE_CHECKING:
     from vllm.worker.model_runner import ModelInputForGPUBuilder
 
+from vllm_flash_attn import flash_attn_varlen_func as _flash_attn_varlen_func
+from vllm_flash_attn import flash_attn_with_kvcache as _flash_attn_with_kvcache
+
+
+@torch.library.custom_op("vllm::flash_attn_varlen_func",
+                         mutates_args=[
+                             "q", "k", "v", "cu_seqlens_q", "cu_seqlens_k",
+                             "block_table"
+                         ])
+def flash_attn_varlen_func(
+    q: torch.Tensor,
+    k: torch.Tensor,
+    v: torch.Tensor,
+    cu_seqlens_q: torch.Tensor,
+    cu_seqlens_k: torch.Tensor,
+    max_seqlen_q: int,
+    max_seqlen_k: int,
+    softmax_scale: Optional[float] = None,
+    causal: bool = False,
+    window_size: Optional[List[int]] = None,
+    softcap: float = 0.0,
+    alibi_slopes: Optional[List[float]] = None,
```

Comment:
I'm pretty sure the proper type for this is `Optional[torch.Tensor]`.

Reply:
nice catch! fixed in [495d2f0](https://github.com/vllm-project/vllm/pull/7536/commits/495d2f02703d012f0d8e3047d191c2b9e8c9a521)